packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -355,35 +356,35 @@ class AndroidDevice extends Device {
       printTrace('Waiting for observatory port to be available...');
 
       try {
-        int observatoryDevicePort, diagnosticDevicePort;
```

Comment:
We don't need iOS and Android specific versions of this code, this whole try block is actually common code and should be refactored into a shared helper which returns two uris.